cohortextractor2/docs.py

```diff
@@ -0,0 +1,62 @@
+import json
+import operator
+
+from .backends.base import BaseBackend
+from .concepts.tables import ClinicalEvents, PatientDemographics, PracticeRegistrations
+
+
+def _build_backends():
+    backends = sorted(BaseBackend.__subclasses__(), key=operator.attrgetter("__name__"))
+
+    for backend in backends:
+        tables = [getattr(backend, name) for name in backend.tables]
+        tables = [table.implements.__name__ for table in tables if table.implements]
+        yield {
+            "name": backend.__name__,
+            "tables": tables,
+        }
+
+
+def _build_contracts():
+    """Build a dict representation for each Contract"""
+    # TODO: investigate using TableContract.__subclasses__() to build this list
+    # dynamically
+    contracts = [
+        ClinicalEvents,
+        PatientDemographics,
+        PracticeRegistrations,
+    ]
+
+    for contract in contracts:
+        docstring = _reformat_docstring(contract.__doc__)
+        dotted_path = f"{contract.__module__}.{contract.__qualname__}"
+
+        yield {
+            "name": contract.__name__,
+            "dotted_path": dotted_path,
+            "docstring": docstring,
+            "columns": [],
+            "contract_support": [],
+        }
+
+
+def _reformat_docstring(d):
+    """Reformat docstring to make it easier to use in a markdown/HTML document."""
+    docstring = d.strip()
+
+    lines = (line.strip() for line in docstring.split("\n"))
+    lines = (line for line in lines if line)  # remove empty items
```

Comment:
Isn't there an argument for allowing paragraphs here?

Reply:
I'm not against them!  It just looked odd to me to maintain the formatting of the docstrings for the docs, do you have a preference?

Reply:
Of course we can't really know until we've got a bunch of examples. But if the person writing the docstring feels the need for multiple paragraphs then I think it seems likely we'll want them in the final output.

You could make it a list of strings rather than relying on semantic empty lines.

Reply:
Yep that would work too.  This was to avoid

```
First line.

Second line.
```

getting turned into

```First line.  Second line.```

Note the double space between sentences!

Reply:
Fixed and rebased.